wire-ios/Wire-iOS Tests/ConversationList/Container/ConversationListViewControllerTests.swift

```diff
@@ -109,6 +110,94 @@ final class ConversationListViewControllerTests: XCTestCase {
         verify(matching: tabBarController)
     }
 
+    // MARK: - Snapshot Tests for Filter View
+
+    func testForShowingConversationsWithoutAnyFilterApplied() {
+        // GIVEN
+        let conversations = createConversations(
+            names: ["iOS Team", "Web Team", "QA Team", "Design Team", "iOS Bugs & Questions"]
+        )
+        userSession.mockConversationDirectory.mockUnarchivedConversations = conversations
+
+        // WHEN
+        sut.hideNoContactLabel(animated: false)
+        sut.applyFilter(nil)
+
+        // THEN
+        verify(matching: tabBarController)
+    }
+
+    func testForShowingConversationsFilteredByGroups() {
+        // GIVEN
+        let conversations = createConversations(names: ["iOS Team", "Web Team"])
+        userSession.mockConversationDirectory.mockGroupConversations = conversations
+
+        // WHEN
+        sut.hideNoContactLabel(animated: false)
+        sut.applyFilter(.groups)
+
+        // THEN
+        verify(matching: tabBarController)
+    }
+
+    func testForShowingConversationsFilteredByFavourites() {
+        // GIVEN
+        let conversations = createConversations(names: ["iOS Team", "Web Team"], favorite: [true, false])
+        userSession.mockConversationDirectory.mockFavoritesConversations = conversations.filter { $0.isFavorite }
+
+        // WHEN
+        sut.hideNoContactLabel(animated: false)
+        sut.applyFilter(.favorites)
+
+        // THEN
+        verify(matching: tabBarController)
+    }
+
+    func testForShowingConversationsFilteredByOneOnOne() throws {
+        throw XCTSkip("We can't really work make this test work until we refactor all of the code we have related to User Types")
+
+        // GIVEN
+        let modelHelper = ModelHelper()
+        let fixture = CoreDataFixture()
+
+        let user1 = modelHelper.createUser(in: fixture.coreDataStack.viewContext)
+        let user2 = modelHelper.createUser(in: fixture.coreDataStack.viewContext)
+
+        let oneOnOneConversation1 = modelHelper.createOneOnOne(with: user1, in: fixture.coreDataStack.viewContext)
+        let oneOnOneConversation2 = modelHelper.createOneOnOne(with: user2, in: fixture.coreDataStack.viewContext)
+
+        coreDataFixture.coreDataStack.viewContext.conversationListDirectory().refetchAllLists(
+            in: coreDataFixture.coreDataStack.viewContext
+        )
+
+        userSession.mockConversationDirectory.mockContactsConversations = [oneOnOneConversation1, oneOnOneConversation2]
+
+        // WHEN
+        sut.hideNoContactLabel(animated: false)
+        sut.applyFilter(.oneToOneConversations)
+
+        // THEN
+        verify(matching: tabBarController)
+    }
+
+    // MARK: - Helper Methods
+
+    private func createConversations(names: [String], favorite: [Bool] = []) -> [ZMConversation] {
```

Comment:
suggestion: pass in a tuple `(String, Bool)` so you can iterate over pairs of values together rather than using indices:

```swift
for (name, isFavorite) in {
  ...
}